I tested running `python check_style.py` from the command line and got following errors:

``` bash
INFO:__main__:Checking E:\projects\seattle-leetcoders\InterviewProblems\check_style.py
Traceback (most recent call last):
  File "check_style.py", line 23, in check_style
    subprocess.check_output(['pylint', module])
  File "C:\Users\zhzex\AppData\Local\Programs\Python\Python35-32\lib\subprocess.py", line 626, in check_output
    **kwargs).stdout
  File "C:\Users\zhzex\AppData\Local\Programs\Python\Python35-32\lib\subprocess.py", line 708, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['pylint', 'E:\\projects\\seattle-leetcoders\\InterviewProblems\\check_style.py']' returned non-zero exit status 16

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "check_style.py", line 99, in <module>
    main()
  File "check_style.py", line 83, in main
    pylint_runner.check_style(file_path)
  File "check_style.py", line 30, in check_style
    linter_message).group(0)
  File "C:\Users\zhzex\AppData\Local\Programs\Python\Python35-32\lib\re.py", line 173, in search
    return _compile(pattern, flags).search(string)
TypeError: cannot use a string pattern on a bytes-like object
```

Used a virtualenv with following packages installed:

``` bash
>pip list
astroid (1.4.8)
colorama (0.3.7)
isort (4.2.5)
lazy-object-proxy (1.2.2)
mccabe (0.5.2)
pep8 (1.7.0)
pip (8.1.1)
py (1.4.31)
pylint (1.6.4)
pytest (3.0.1)
setuptools (20.10.1)
six (1.10.0)
wrapt (1.10.8)
```

--- a/check_style.py
+++ b/check_style.py
@@ -1,58 +1,17 @@
 import logging
-import re
 import os
-import subprocess
 import sys
+
+import python_style_checker
 
 
 logging.basicConfig(level=logging.INFO)
 LOGGER = logging.getLogger(__name__)
 
+LINTERS = [python_style_checker.PylintRunner(),
+           python_style_checker.Pep8Runner()]
+
 IGNORED_FILES = set()
-
-
-class PylintRunner(object):
-
-    def __init__(self):
-        self.passed = 0
-        self.failed = 0
-
-    def check_style(self, module):
-        '''Runs pylint on a Python module.'''
-        try:
-            subprocess.check_output(['pylint', module])
-            LOGGER.debug('%s PASSED.', module)
-            self.passed += 1
-        except subprocess.CalledProcessError as pylint_error:
-            linter_message = pylint_error.output
-            try:
-                score = re.search(r'[-0-9]+\.\d{2}/10',
-                                  linter_message).group(0)
-                LOGGER.warning('%s failed PYLINT with score %s.',
-                               os.path.relpath(module), score)
-                self.failed += 1
-            except AttributeError:
-                raise
-
-
-class Pep8Runner(object):
-
-    def __init__(self):
-        self.passed = 0
-        self.failed = 0
-
-    def check_style(self, module):
-        '''Runs pep8 on a Python module.'''
-        try:
-            subprocess.check_output(['pep8', module])
-            LOGGER.debug('%s PASSED.', module)
-            self.passed += 1
-        except subprocess.CalledProcessError as pep8_error:
-            pep8_message = pep8_error.output
-            LOGGER.warning('%s FAILED PEP8', os.path.relpath(module))
-            self.failed += 1
-            for pep8_warning in pep8_message.splitlines():
-                LOGGER.warning(pep8_warning)
 
 
 def _should_check(module):
@@ -68,30 +27,27 @@ def _get_starting_directory(args):
     return base_directory
 
 
-def main():
-    base_directory = _get_starting_directory(sys.argv)
-    LOGGER.debug('Inspecting in directory %s...', base_directory)
-
-    pylint_runner = PylintRunner()
-    pep8_runner = Pep8Runner()
-
+def _run_linter(base_directory, linter):
+    LOGGER.info('***************Running %s***************', linter.linter_name)
     for root, _, files in os.walk(base_directory):
         for name in files:
             file_path = os.path.join(root, name)
             if _should_check(file_path):
                 LOGGER.info('Checking %s', file_path)
-                pylint_runner.check_style(file_path)
-                pep8_runner.check_style(file_path)
+                linter.check_style(file_path)
             else:
                 LOGGER.debug('Ignore %s', file_path)
+    LOGGER.info('%s:\n\tPassed: %d;\n\tFailed: %d', linter.linter_name,
+                linter.passed, linter.failed)
 
-    LOGGER.info('PYLINT:\n\tPassed: %d;\n\tFailed: %d', pylint_runner.passed,
-                pylint_runner.failed)
 
-    LOGGER.info('PEP8:\n\tPassed: %d;\n\tFailed: %d', pep8_runner.passed,
-                pep8_runner.failed)
+def main():
+    base_directory = _get_starting_directory(sys.argv)
+    LOGGER.debug('Inspecting in directory %s...', base_directory)
+    for linter in LINTERS:
+        _run_linter(base_directory, linter)
 
-    if pylint_runner.failed > 0 or pep8_runner.failed > 0:
+    if any(linter.failed > 0 for linter in LINTERS):
         sys.exit()
 
 

--- a/python_style_checker.py
+++ b/python_style_checker.py
@@ -1,0 +1,61 @@
+import logging
+import os
+import re
+import subprocess
+
+
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+
+
+class PythonStyleChecker(object):
+    def __init__(self, linter_name):
+        self.passed = 0
+        self.failed = 0
+        self._linter_name = linter_name
+
+    @property
+    def linter_name(self):
+        return self._linter_name
+
+    def check_style(self, module):
+        raise NotImplementedError
+
+
+class PylintRunner(PythonStyleChecker):
+
+    def __init__(self):
+        super(PylintRunner, self).__init__('pylint')
+
+    def check_style(self, module):
+        '''Runs pylint on a Python module.'''
+        try:
+            subprocess.check_output(['pylint', module])
+            LOGGER.debug('%s PASSED.', module)
+            self.passed += 1
+        except subprocess.CalledProcessError as pylint_error:
+            linter_message = pylint_error.output
+            score_found = re.search(br'[-0-9]+\.\d{2}/10', linter_message)
+            score = score_found.group(0)
+            LOGGER.warning('%s failed PYLINT with score %s.',
+                           os.path.relpath(module), score)
+            self.failed += 1
+
+
+class Pep8Runner(PythonStyleChecker):
+
+    def __init__(self):
+        super(Pep8Runner, self).__init__('pep8')
+
+    def check_style(self, module):
+        '''Runs pep8 on a Python module.'''
+        try:
+            subprocess.check_output(['pep8', module])
+            LOGGER.debug('%s PASSED.', module)
+            self.passed += 1
+        except subprocess.CalledProcessError as pep8_error:
+            pep8_message = pep8_error.output
+            LOGGER.warning('%s FAILED PEP8', os.path.relpath(module))
+            self.failed += 1
+            for pep8_warning in pep8_message.splitlines():
+                LOGGER.warning(pep8_warning)

--- a/python_style_checker.py
+++ b/python_style_checker.py
@@ -30,7 +30,7 @@ class PylintRunner(PythonStyleChecker):
     def check_style(self, module):
         '''Runs pylint on a Python module.'''
         try:
-            subprocess.check_output(['pylint', module])
+            subprocess.check_output([self.linter_name, module])
             LOGGER.debug('%s PASSED.', module)
             self.passed += 1
         except subprocess.CalledProcessError as pylint_error:
@@ -50,7 +50,7 @@ class Pep8Runner(PythonStyleChecker):
     def check_style(self, module):
         '''Runs pep8 on a Python module.'''
         try:
-            subprocess.check_output(['pep8', module])
+            subprocess.check_output([self.linter_name, module])
             LOGGER.debug('%s PASSED.', module)
             self.passed += 1
         except subprocess.CalledProcessError as pep8_error:

